code/client/src/components/AddressInput.jsx

```diff
@@ -24,13 +34,53 @@ const AddressInput = ({ setAddressCallback, currentWallet, addressValue, extraSe
   const { isMobile } = useWindowDimensions()
 
   const deleteKnownAddress = useCallback((address) => {
-    setAddressCallback({ value: '' })
+    setAddressCallback({ value: '', label: '' })
     dispatch(walletActions.deleteKnownAddress(address))
   }, [dispatch])
 
-  const onSearchAddress = useCallback((address) => {
-    setAddressCallback({ value: address, label: address })
-  }, [setAddressCallback])
+  const onSearchAddress = (value) => {
+    setSearchingAddress(true)
+    setSearchValue(value)
+  }
+
+  useWaitExecution(
+    async () => {
+      try {
+        const validAddress = util.safeNormalizedAddress(searchValue)
+
+        if (validAddress) {
+          const domainName = await api.blockchain.domain.reverseLookup({ address: validAddress })
+
+          setAddressCallback({
+            value: validAddress,
+            domainName,
+            filterValue: searchValue,
+            selected: false
+          })
+        }
+
+        if (!isEmpty(searchValue)) {
```

Comment:
could do some preliminary filtering to reduce queries to blockchain, e.g. `searchValue.includes(`${ONEConstants.Domain.DEFAULT_PARENT_LABEL}.${ONEConstants.Domain.DEFAULT_TLD} )`